integration-tests/src/test/java/tachyon/master/MasterClientAuthenticationIntegrationTest.java

```diff
@@ -0,0 +1,192 @@
+
+  @Before
+  public void before() throws Exception {
+    mLocalTachyonCluster = new LocalTachyonCluster(1000, 1000, Constants.GB);
+    mExecutorService = Executors.newFixedThreadPool(2);
+    clearLoginUser();
+  }
+
+  @After
+  public void after() throws Exception {
+    System.clearProperty(Constants.TACHYON_SECURITY_AUTHENTICATION);
+    System.clearProperty(Constants.TACHYON_AUTHENTICATION_PROVIDER_CUSTOM_CLASS);
+    System.clearProperty(Constants.TACHYON_SECURITY_USERNAME);
+  }
+
+  @Test
+  public void noAuthenticationOpenCloseTest() throws Exception {
+    //no authentication configure
+    System.setProperty(Constants.TACHYON_SECURITY_AUTHENTICATION,
+        AuthType.NOSASL.getAuthName());
+    //start cluster
+    mLocalTachyonCluster.start();
+
+    authenticationOperationTest("/file-nosasl");
+
+    //stop cluster
+    mLocalTachyonCluster.stop();
+  }
+
+  @Test
+  public void simpleAuthenticationOpenCloseTest() throws Exception {
+    //simple authentication configure
+    System.setProperty(Constants.TACHYON_SECURITY_AUTHENTICATION,
+        AuthType.SIMPLE.getAuthName());
+    //start cluster
+    mLocalTachyonCluster.start();
+
+    authenticationOperationTest("/file-simple");
+
+    //stop cluster
+    mLocalTachyonCluster.stop();
+  }
+
+  @Test
+  public void customAuthenticationOpenCloseTest() throws Exception {
+    //custom authentication configure
+    System.setProperty(Constants.TACHYON_SECURITY_AUTHENTICATION,
+        AuthType.CUSTOM.getAuthName());
+    //custom authenticationProvider configure
+    System.setProperty(Constants.TACHYON_AUTHENTICATION_PROVIDER_CUSTOM_CLASS,
+        NameMatchAuthenticationProvider.class.getName());
+
+    /**
+     * Using tachyon as loginUser for unit testing, only tachyon user is allowed to connect to
+     * Tachyon Master.
+     */
+    System.setProperty(Constants.TACHYON_SECURITY_USERNAME, "tachyon");
+
+    //start cluster
+    mLocalTachyonCluster.start();
+
+    authenticationOperationTest("/file-custom");
+
+    //stop cluster
+    mLocalTachyonCluster.stop();
+  }
+
+  @Test
+  public void customAuthenticationDenyConnectTest() throws Exception {
+    //custom authentication configure
+    System.setProperty(Constants.TACHYON_SECURITY_AUTHENTICATION,
+        AuthType.CUSTOM.getAuthName());
+    //custom authenticationProvider configure
+    System.setProperty(Constants.TACHYON_AUTHENTICATION_PROVIDER_CUSTOM_CLASS,
+        NameMatchAuthenticationProvider.class.getName());
+    /**
+     * Using tachyon as loginUser for unit testing, only tachyon user is allowed to connect to
+     * Tachyon Master.
+     */
+    System.setProperty(Constants.TACHYON_SECURITY_USERNAME, "tachyon");
+    //start cluster
+    mLocalTachyonCluster.start();
+
+    mMasterInfo = mLocalTachyonCluster.getMasterInfo();
+    // Using no-tachyon as loginUser to connect to Master, the IOException will be thrown
+    clearLoginUser();
+    mThrown.expect(IOException.class);
```

Comment:
Because the real exception was wrapped by the IOException from MasterClient. So I can't test the exception message here. Thanks for your comment.